src/gui/pages_menu/KM_GUIMenuLoad.pas

```diff
@@ -159,19 +191,22 @@ procedure TKMMenuLoad.Load_Delete_Click(Sender: TObject);
   begin
     PreviouslySelected := ColumnBox_Load.ItemIndex;
     fSaves.DeleteSave(ColumnBox_Load.ItemIndex);
-    Load_RefreshList(False);
-    if ColumnBox_Load.RowCount > 0 then
-      ColumnBox_Load.ItemIndex := EnsureRange(PreviouslySelected, 0, ColumnBox_Load.RowCount - 1)
-    else
-      ColumnBox_Load.ItemIndex := -1; //there are no saves, nothing to select
-    Load_ListClick(ColumnBox_Load);
+
+    if ColumnBox_Load.RowCount > 1 then
+    begin
+      NewSelected := EnsureRange(PreviouslySelected, 0, ColumnBox_Load.RowCount - 2);
+      SetLastSaveCRC(fSaves[NewSelected].CRC);
+    end else
+      SetLastSaveCRC; //there are no saves, nothing to select
+
+    Load_RefreshList(True);
   end;
 end;
 
 
 procedure TKMMenuLoad.Load_ScanUpdate(Sender: TObject);
 begin
-  Load_RefreshList(False); //Don't jump to selected with each scan update
+  Load_RefreshList(True); //Jump to selected with each scan update
```

Comment:
I checked it with debug variable and its very unpleasant. 
The reason why I set it to true is because I want to jump to selected after scan finished. For example on MapEd menu open or when switching map type.
So I think the proper solution will be jump to selected on scan complete.
I will refactor code due to this change